### PR TITLE
[script/pull_github_pr.sh] add `--force` option when gating is not stable

### PR DESCRIPTION
**Pull request by yaronkaikov**
As a follow-up for ec0bb42b45e1d15b9d8b5fbc861593fadf5c98a3, in case gating is not stable and the maintainer needs to merge patches anyway, adding `--force` to override it

Also improving the message when `JENKINS_API_TOKEN` is missing from env

** merge script improvement, no need for backport **

### CODE REVIEW

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:22]: "enva"?

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:23]: The sentence just ends in the middle. Needs a final point, and also maybe say something like "and set the JENKINS_USERNAME and JENKINS_API_TOKEN environment variables accordingly"?

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:62]: Why is this no longer needed? After all, you can also set your environment variables incorrectly!
But please replace the phrase "Please check your session or credentials" (what does this even mean - what's a "session"?) to please check your JENKINS_USERNAME and JENKINS_API_TOKEN setting.

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:77]: Nitpick: would be nice to get a link to the job which you consider unstable. As a maintainer, I might want to take a look into that, but not remember exactly where that job is.

By the way, I hope that your script actually works on various branches (open-source, enterprise, different branches, etc.) and NOT JUST ONE MASTER. We use the same script on all branches... If it doesn't work on other branches, another followup will be necessary.

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:21]: nitpick, this is a shell, not Python, so triple-quote is meaningless - actually the first two quotes are an empty string and then you have one more real quote. So the triple quotes is just the same as one quotes.

Note that using quotes the way you did will print out potentially-unwanted indentation as part of the message. The traditional shell trick to print a long message without indentation is a so-called "**here document**":

```bash
cat <<-END
<tab>hello
<tab>hi
END
```
but sadly, this was invented in the days that only tabs were used for indentation so it doesn't work with spaces so it's worthless for us.

File: scripts/pull_github_pr.sh
Comment (yaronkaikov) [reply scripts/pull_github_pr.sh]: typo, fixed

File: scripts/pull_github_pr.sh
Comment (yaronkaikov) [reply scripts/pull_github_pr.sh]: Fixed

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:70]: This code looks completely wrong to me - unless I'm misunderstanding something.
First you removed (by mistake?) the "exit 1".
Then, you check the URL of the request (!) instead of its result??

Please make sure you actually test (manually, I'm not expecting anything fancier) each feature you add in a scenario where it is reached!

File: scripts/pull_github_pr.sh
Comment (yaronkaikov) [reply scripts/pull_github_pr.sh]: it will work on master and OSS releases. Once this gets merged to `enterprise` I will modify it for enterprise releases. But I assume we are not going to backport this, right?

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:81]: I think you shouldn't print this message if "--force" is already active. Perhaps print a different message saying the build is unstable but don't say "exiting" or recommend using "--force".

File: scripts/pull_github_pr.sh
Comment (nyh) [inline scripts/pull_github_pr.sh:82]: This means that the second parameter "dog" will do the same thing as "--force". Please check just for "--force".

**Review by nyh — CHANGES_REQUESTED** [review]

File: scripts/pull_github_pr.sh
Comment (nyh) [reply scripts/pull_github_pr.sh]: I was under the impression (but maybe @avikivity can correct me I'm wrong) that the enterprise script is identical, it just recognizes that the current working directory is an enterprise working directory. If we start having N different versions of this script, it will just make our life harder going forward any time we need to improve this script...

Backports is not the important issue - consider in a year's time this same script will be in branch-7.0, branch-7.1, enterprise-2025.1, and so on - and it will need to work correctly. Instead of hacking it separately on every branch it can - and probably should - recognize in which branch it is. I think this is what this script has been doing so far (but @avikivity can correct me if I'm wrong).

Comment (nyh) [reply scripts/pull_github_pr.sh]: By the way, I work with exactly the same script also to merge Seastar patches - because also in Seastar we use the same merge process! It is really sad to see that won't work any more because of some irrelevant Jenkins verification.
@avikivity what script do you use to merge Seastar patches?

**Comment by yaronkaikov** [comment]
@nyh https://github.com/scylladb/scylladb/commit/ec0bb42b45e1d15b9d8b5fbc861593fadf5c98a3 was reverted by @denesb , i will re-work  on this and re-submit this patch

**Comment by yaronkaikov** [comment]
Closing this in favor of https://github.com/scylladb/scylladb/pull/20742 , all comments will be address there

**Pull request closed without merging**
